content/developers/developer-patterns/scitt-api/index.md

```diff
@@ -0,0 +1,164 @@
+date: 2021-06-09T13:49:35+01:00
+lastmod: 2021-06-09T13:49:35+01:00
+draft: false
+images: []
+menu:
+  developers:
+    parent: "developer-patterns"
+weight: 112
+toc: true
+aliases: 
+  - /docs/developer-patterns/scitt-api/
+---
+
+{{< caution >}}
+The SCITT API is currently in preview and subject to change
+{{< /caution >}}
+
+The **S**upply **C**hain **I**ntegrity, **T**ransparency and **T**rust (SCITT) initiative is a set of [IETF standards](https://datatracker.ietf.org/group/scitt/documents/) for managing the compliance and auditability of goods and services across end-to-end supply chains.
+SCITT supports the ongoing verification of goods and services where the authenticity of entities, evidence, policy, and artifacts can be assured and the actions of entities can be guaranteed to be authorized, non-repudiable, immutable, and auditable.
+
+To assure insights to supply chain artifacts are current, the SCITT APIs provide a correlation of statements for a specific artifact, allowing newer information to be registered for the most up to date insights.
+
+This quickstart will:
+
+1. create, or use an existing a key to sign a collection of statements about an artifact
+1. create and register an SBOM for the artifact
+1. create and register an attestation for the artifact
+1. query a collection of statements about the artifact
+
+## Prerequisites
+
+- [A DataTrails subscription](https://app.datatrails.ai/signup)
+- [DataTrails sample code](#datatrails-sample-code)
+- [SBOM Tool](https://github.com/microsoft/sbom-tool)
+
+### DataTrails Sample Code
+
+Clone the [DataTrails SCITT Examples](https://github.com/datatrails/datatrails-scitt-samples) repository
+
+```bash
+git clone https://github.com/datatrails/datatrails-scitt-samples.git
+
+cd datatrails-scitt-samples
+```
+
+## Create a Signing Key
+
+{{< note >}}
+If you already have a COSE Key, skip ahead to [Register a SBOM for the Artifact](#register-a-sbom-for-the-artifact)
+{{< /note >}}
+
+DataTrails supports multiple signing keys.
+_(TODO: link to supported DataTrails Signing Keys\)_<br>
+For a quickstart, we'll create a testing [COSE Key](https://cose-wg.github.io/cose-spec/#key-structure) which DataTrails will cryptographically validate upon registration
+
+1. Create a local signing key
+
+    ```shell
+    openssl ecparam -name prime256v1 -genkey -out scitt-signing-key.pem
+    ```
+
+## Register a SBOM for the Artifact
+
+When registering statements about an artifact, a common identifier is required to correlate a collection of statements.
+In SCITT, a `CWT_Claims subject` property captures the identifier.
+
+The example generates an [SPDX SBOM](https://spdx.dev/), which generates a [Document Namespace](https://spdx.github.io/spdx-spec/v2.2.2/document-creation-information/#65-spdx-document-namespace-field) field which is used for the `CWT_Claims subject` property.
+
+_\<TODO: Add a doc for creating unique identifiers>_
```

Comment:
I don't think we need super unique identifiers.

The subject can be an arbitrary string? something simple for the sbom example `my software`?

Reply:
I think we want to articulate that it is the issuers choice entirely. So using 'my software' is fine for the sake of having the simplest possible example. But we know the reason people are signing statements is to promote trust from others. So they will need better identifiers than just 'my software' to achieve that. I think it would help if we could at least point at one or two concrete examples of statement id schemes (assuming something exists), and if none do then we should propose something but make it clear it is an example

Reply:
Following statement ID schemes would be great! Even if it is a made up :)

Reply:
I've updated it to say identifiers. It's just a placeholder for now.